drivers/sensor/tsl2520/tsl2520.c

```diff
@@ -0,0 +1,220 @@
+
+#define LO_BYTE(X)	(X & 0xFF)
+#define HI_BYTE(X)	((X>>8) & 0xFF)
+
+LOG_MODULE_REGISTER(tsl2520, CONFIG_SENSOR_LOG_LEVEL);
+
+static int tsl2520_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct tsl2520_config *cfg = dev->config;
+	struct tsl2520_data *data = dev->data;
+	int ret = 0;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT ||
+			chan == SENSOR_CHAN_IR);
+
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT || chan == SENSOR_CHAN_IR) {
+		uint8_t read_low_byte;
+		uint8_t read_high_byte;
+		uint8_t reg_status;
+
+		/* In order to update ALS Data Registers ALS_STATUS must be read first */
+		(void)i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_STATUS, &reg_status);
+
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_STATUS, &reg_status);
+		if (ret < 0) {
+			LOG_ERR("Failed reading chip status");
+			return -EIO;
+		}
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_ALS_DATA0_LOW, &read_low_byte);
+		if (ret < 0) {
+			LOG_ERR("Could not fetch als sensor value");
+			return -EIO;
+		}
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_ALS_DATA0_HIGH, &read_high_byte);
+		if (ret < 0) {
+			LOG_ERR("Could not fetch als sensor value");
+			return -EIO;
+		}
+		data->als_data0 = (read_high_byte << 8) | read_low_byte;
+		LOG_DBG("data0 value: %X ",data->als_data0 );
+
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_ALS_DATA1_LOW, &read_low_byte);
+		if (ret < 0) {
+			LOG_ERR("Could not fetch als sensor value");
+			return -EIO;
+		}
+		ret = i2c_reg_read_byte_dt(&cfg->i2c_spec, TSL2520_REG_ALS_DATA1_HIGH, &read_high_byte);
+		if (ret < 0) {
+			LOG_ERR("Could not fetch als sensor value");
+			return -EIO;
+		}
+		data->als_data1 = (read_high_byte << 8) | read_low_byte;
+		LOG_DBG("data1 value: %X ",data->als_data1 );
+
+		if (ret < 0) {
+			LOG_ERR("Could not fetch ambient light (visible)");
+			return -EIO;
+		}
+	}
+	
+	return ret;
+}
+
+static int tsl2520_channel_get(const struct device *dev, enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct tsl2520_data *data = dev->data;
+	int ret = 0;
+	if (chan == SENSOR_CHAN_ALL || chan == SENSOR_CHAN_LIGHT || chan == SENSOR_CHAN_IR) {
+		val->val1 = data->als_data0;
+		val->val2 = data->als_data1;
+		return ret;
+	}
+
+	return -ENOTSUP;
+}
+
+static int tsl2520_setup(const struct device *dev)
+{
+	const struct tsl2520_config *cfg = dev->config;
+	int ret;
+	uint8_t tmp;
+	uint16_t temp;
+	ret = i2c_reg_write_byte_dt(&cfg->i2c_spec, TSL2520_ENABLE_ADDR, TSL2520_ENABLE_AEN_PON);
+	if (ret < 0) {
+		LOG_ERR("Failed to enable register");
+		return ret;
+	}
+
+	tmp = cfg->sample_time;
+	ret = i2c_reg_write_byte_dt(&cfg->i2c_spec, TSL2520_SAMPLE_TIME0, tmp);
+	if (ret < 0) {
+		LOG_ERR("Failed setting sample time");
+		return ret;
+	}
+
+	temp = cfg->als_nr_samples;
+	ret = i2c_reg_write_byte_dt(&cfg->i2c_spec, TSL2520_ALS_NR_SAMPLES0, LO_BYTE (temp));
```

Comment:
Skip LO_BYTE/HI_BYTE and use i2c_burst_write_dt() instead.
The device supports burst writes and even recommends it.

"All 16-bit fields have a latching scheme for reading and writing.
In general it is recommended to use I²C bursts whenever
possible, "

- https://ams.com/documents/20143/4410897/TSL2520_DS001068_3-00.pdf